[Feature] Add 'Configuring unique connections' blog post

gwenwindflower opened the pull request:
## What are you changing in this pull request and why?

This PR adds a blog post covering Extended Attributes and some other related features for configuring more unique Environment and User connections in dbt Cloud.

gwenwindflower committed:
Spike out article and author for Gwen Windflower

gwenwindflower committed:
Remove bun lockfile

gwenwindflower committed:
Add images to EAT article

gwenwindflower committed:
Update draft of EAT blog

gwenwindflower committed:
Merge branch 'current' into feat/add-eat-blog

gwenwindflower committed:
Update EAT blog with Florian's feedback

gwenwindflower committed:
Merge branch 'feat/add-eat-blog' of github.com:dbt-labs/docs.getdbt.com into feat/add-eat-blog

gwenwindflower committed:
Merge branch 'current' into feat/add-eat-blog

ericbaumann reviewed (commented):
Purely from a content standpoint, this looks good to me. 

From an organization perspective, i might suggest delineating the content a little more between what I observe as 3 fairly distinct features that allow cloud to be more flexible in ways that meet user needs.

1. EAT & its applications
2. The staging deployment type, why it's cool, and what it allows you to do
3. User develop overrides and a scenario where you might reach for it

ericbaumann commented:
`features that allow cloud to be more flexible in ways that meet user needs.`
and i'm interpreting this _roughly_ as the overall intent of the article

gwenwindflower commented:
@ericbaumann yea that's right, sorry it's confusing that i was referring to it as an EAT blog, just was originally scoped as that and stuck in my head that way, but yes, one of the pivots was understanding that the ask was really more to cover these features and how they have a common theme around enabling unique connections/environments in new ways. I'll do a pass at making that more clear! Thanks!

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+is_featured: true
+---
+
+dbt Cloud has a suite of new features that enable configuring precise and unique connections to data platforms at the Environment and User level. These enable more sophisticated setups, like connecting a project to multiple warehouse accounts, first class support for Staging environments, and User-level overrides for specific dbt versions. This gives dbt Cloud developers the features they need to ...
```
suggs:
- present tense
- links for deeper dives
- italics for emphasis

```suggestion
dbt Cloud now includes a suite of new features that enable configuring precise and unique connections to data platforms at the environment and user level. These enable more sophisticated setups, like connecting a project to multiple warehouse accounts, first-class support for [staging environments](/docs/deploy/deploy-environments#staging-environment), and user-level [overrides for specific dbt versions](/docs/dbt-versions/upgrade-dbt-version-in-cloud#override-dbt-version). This gives dbt Cloud developers the features they need to tackle more complex tasks, like Write-Audit-Publish (WAP) workflows and safely testing dbt version upgrades. While you still configure a default connection at the project level and per-developer, you now have tools to get more advanced in a secure way. Soon, dbt Cloud will take this even further allowing multiple connections to be set globally and reused with _global connections_.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+<!--truncate-->
+
+The first new feature we’re going to look at is called Extended Attributes.
```
```suggestion
The first new feature we’re going to look at is called [extended attributes](/docs/dbt-cloud-environments#extended-attributes).
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## Profile pick
+
+Extended Attributes are a tool that brings the flexibility of dbt Core’s `profiles.yml` configuration to dbt Cloud. Before Extended Attributes, you configured a project-level connection, and were mostly stuck with it. You could develop and orchestrate into different schemas to keep development work away from production, or configure a staging layer with manual workarounds, but beyond that things  ...
```
```suggestion
Extended attributes is a feature that brings the flexibility of dbt Core’s `profiles.yml` configuration to dbt Cloud. Before the release of the extended attributes feature, you configured a project-level connection and were mostly stuck with it. You could develop and orchestrate into different schemas to keep development work away from production or configure a staging layer with manual workarounds but, beyond that, things got more challenging. By borrowing the flexibility of `profiles.yml`, which allows configuring as many unique connections as you need, you can now do the same with the security and orchestration tools in dbt Cloud.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+Extended Attributes are a tool that brings the flexibility of dbt Core’s `profiles.yml` configuration to dbt Cloud. Before Extended Attributes, you configured a project-level connection, and were mostly stuck with it. You could develop and orchestrate into different schemas to keep development work away from production, or configure a staging layer with manual workarounds, but beyond that things  ...
+
+## How Extended Attributes Work
```
sugg: sentence casing vs title casing

```suggestion
## How extended attributes work
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## How Extended Attributes Work
+
+Extended Attributes are a textbox on the Environment settings page, where you can input `profiles.yml` type configurations. When developing in the dbt Cloud IDE, dbt Cloud CLI, or orchestrating job runs, dbt Cloud will parse through the provided Extended Attributes YAML and merge it with your base project connection settings. If the attribute exists in another source (at present this would typica ...
```
```suggestion
The **Extended attributes** option is available as a textbox on the **Environment settings** page, where you can input `profiles.yml` type configurations. When developing in the dbt Cloud IDE, dbt Cloud CLI, or orchestrating job runs, dbt Cloud will parse the provided YAML for extended attributes and merge it with your base project connection settings. If the attribute exists in another source (typically, this would be your project connection settings or the job's configurations), it will _replace_ its value, including overriding any custom environment variables. If the attribute doesn't exist, it will add the attribute to the connection config. You [can check out the documentation](https://docs.getdbt.com/docs/deploy/deploy-environments#extended-attributes) for more specific details, but now that you’ve got the basic idea, let’s dive into some examples to see why this is so cool.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+Extended Attributes are a textbox on the Environment settings page, where you can input `profiles.yml` type configurations. When developing in the dbt Cloud IDE, dbt Cloud CLI, or orchestrating job runs, dbt Cloud will parse through the provided Extended Attributes YAML and merge it with your base project connection settings. If the attribute exists in another source (at present this would typica ...
+
+## Multiple accounts for Development and Production environments
```
sugg: sentence casing

```suggestion
## Multiple accounts for development and production environments
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## Multiple accounts for Development and Production environments
+
+The most pressing use case for dbt Cloud users is the ability to use different account connections for different teams or development stages in their pipelines. Let’s consider a team that has a typical dev, staging, production setup (known as a Write-Audit-Publish [WAP] workflow): development for active work with small datasets, staging to promote and vet changes against cloned production data, a ...
```
sugg: since the first instance of the long form Write-Audit-Publish and the acronym WAP is used earlier on the page, we can simply use the acronym here 

```suggestion
The most pressing use case for dbt Cloud users is the ability to use different account connections for different teams or development stages in their pipelines. Let’s consider a team that has a typical dev, staging, production setup (known as a WAP workflow): development for active work with small datasets, staging to promote and vet changes against cloned production data, and production for the final deployed code that feeds BI tools. For this hypothetical team though, these are separate _accounts_ in their data platform with their own sets of RBAC and other settings. This is a perfect use case for extended attributes. Let’s take a look at how this team might set this up for a company that uses multiple BigQuery accounts, projects, and datasets (projects and datasets are analogous to databases and schemas on other platforms like Snowflake) to separate dev, staging, and prod:
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+The most pressing use case for dbt Cloud users is the ability to use different account connections for different teams or development stages in their pipelines. Let’s consider a team that has a typical dev, staging, production setup (known as a Write-Audit-Publish [WAP] workflow): development for active work with small datasets, staging to promote and vet changes against cloned production data, a ...
+
+![Extended Attributes text box](/img/blog/2024-04-10-extended-attributes/ext_attr.png)
```
i've only used Lightbox for screenshots. for example, like we do here https://github.com/dbt-labs/docs.getdbt.com/blob/5acf48829aa6eed2adb49ccfee3470463beb6cc5/website/docs/docs/deploy/retry-jobs.md?plain=1#L26 

consider updating to (tho blogs may have diff rules)

```suggestion
<Lightbox src="/img/blog/2024-04-10-extended-attributes/ext_attr.png" title="Extended attributes text box" />
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+![Extended Attributes text box](/img/blog/2024-04-10-extended-attributes/ext_attr.png)
+
+_<small>At the bottom of an Environment's Settings you’ll find the Extended Attributes text box.</small>_
```
```suggestion
_<small>At the bottom of an environment's settings you’ll find the **Extended attributes** option.</small>_
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
````diff
@@ -0,0 +1,103 @@
+threads: 16
+```
+
+With this setup, we have a separate account for development work, using individual development datasets for each developer (with a single thread so that the development build logs are easier to read) connected via OAuth, and a shared `staging` project with a default `main` dataset for the Staging environment that is only built to via a GCP Service Account through dbt Cloud. In that project we can ...
````
sugg: since semicolons can be used to list things, sugg using it here for clarity 

```suggestion
With this setup, we have a separate account for development work, using individual development datasets for each developer (with a single thread so that the development build logs are easier to read) connected via OAuth; and a shared `staging` project with a default `main` dataset for the staging environment that's only built via a GCP Service Account through dbt Cloud. In that project, we can then configure IAM permissions to only allow building into the staging schema from jobs that use the staging environment as well.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+With this setup, we have a separate account for development work, using individual development datasets for each developer (with a single thread so that the development build logs are easier to read) connected via OAuth, and a shared `staging` project with a default `main` dataset for the Staging environment that is only built to via a GCP Service Account through dbt Cloud. In that project we can ...
+
+Production is then pointed to a _completely separate account_ that is only writable from Production environment builds and readable from the BI tool.
```
```suggestion
Production is then pointed to a _completely separate account_ that's only writable from production environment builds and readable from the BI tool.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+Production is then pointed to a _completely separate account_ that is only writable from Production environment builds and readable from the BI tool.
+
+It’s really that simple. This works with PrivateLink connections handling the authentication as well! Again, while we have 1 project connection that is the _default_, you can now configure unique connections securely _per Environment_.
```
```suggestion
It’s really that simple. This works with [PrivateLink](/docs/cloud/secure/about-privatelink) connections handling the authentication as well! Again, while we have one project connection that's the _default_, you can now configure unique connections securely _per environment_.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## All the world a Stage
+
+We touched on Staging Environments above in discussing Extended Attributes, but let's dig deeper into how dbt Cloud now supports those in a first class way. You now have the option when configuring an Environment to choose Development, Production, _or_ Staging. When you can configure an Environment as a Staging type, you’ll unlock new abilities, most importantly the ability to defer to _that_ Env ...
```
sugg: if referring to UI labels, apply bold. if not, use lower case and no bolding

```suggestion
Earlier, we touched on staging environments in discussing extended attributes but let's dig deeper into how dbt Cloud now supports those in a first-class way. You now have the option when configuring an environment to choose **Development**, **Production**, _or_ **Staging**. When you configure an environment as a staging type, you’ll unlock new abilities, most importantly the ability to defer to _that_ environment for development work. This fully enables a proper Write-Audit-Publish flow, where development work is built against and promoted to staging before being merged into a production branch when releases have been tested.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+We touched on Staging Environments above in discussing Extended Attributes, but let's dig deeper into how dbt Cloud now supports those in a first class way. You now have the option when configuring an Environment to choose Development, Production, _or_ Staging. When you can configure an Environment as a Staging type, you’ll unlock new abilities, most importantly the ability to defer to _that_ Env ...
+
+All you need to do is configure an Environment as Staging, and toggle the control in the dbt Cloud IDE to “Defer to staging/production”. This will favor a Staging environment over Prod if you have one set up.
```
```suggestion
All you need to do is configure an environment as staging and enable the **Defer to staging/production** option in the dbt Cloud IDE. Doing this will favor a staging environment over prod if you have one set up.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+All you need to do is configure an Environment as Staging, and toggle the control in the dbt Cloud IDE to “Defer to staging/production”. This will favor a Staging environment over Prod if you have one set up.
+
+![Configure environment as Staging](/img/blog/2024-04-10-extended-attributes/env_settings.png)
```
consider updating to use `Lightbox` (see previous comment)

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+All you need to do is configure an Environment as Staging, and toggle the control in the dbt Cloud IDE to “Defer to staging/production”. This will favor a Staging environment over Prod if you have one set up.
+
+![Configure environment as Staging](/img/blog/2024-04-10-extended-attributes/env_settings.png)
+_<small>You can now designate a Staging Environment in a first-class way.</small>_
```
```suggestion
_<small>You can now designate a staging environment in a first-class way.</small>_
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+_<small>You can now designate a Staging Environment in a first-class way.</small>_
+
+![Defer to staging/production](/img/blog/2024-04-10-extended-attributes/defer_to_stage.png)
+_<small>Make sure to toggle on 'Defer to staging/production' to save time and money!</small>_
```
sugg: per style guide, we should apply bold when referring to a UI label

```suggestion
_<small>Make sure to enable **Defer to staging/production** to save time and money!</small>_
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## Upgrading on a curve
+
+Lastly let’s consider a more specialized use case. Imagine we have a ‘tiger team’ (consisting of a lone analytics engineer named Dave) tasked with upgrading from dbt version 1.6 to the new ‘Keep on latest version’ setting, to take advantage of added stability and feature access. We want to keep the rest of the data team being productive in dbt 1.6 for the time being, while enabling Dave to upgrad ...
```
```suggestion
Lastly, let’s consider a more specialized use case. Imagine we have a "tiger team" (consisting of a lone analytics engineer named Dave) tasked with upgrading from dbt version 1.6 to the new **Keep on latest version** setting, to take advantage of added stability and feature access. We want to keep the rest of the data team being productive in dbt 1.6 for the time being, while enabling Dave to upgrade and do his work in the new versionless mode.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+Lastly let’s consider a more specialized use case. Imagine we have a ‘tiger team’ (consisting of a lone analytics engineer named Dave) tasked with upgrading from dbt version 1.6 to the new ‘Keep on latest version’ setting, to take advantage of added stability and feature access. We want to keep the rest of the data team being productive in dbt 1.6 for the time being, while enabling Dave to upgrad ...
+
+### The Development Environment
```
```suggestion
### Development environment
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+### The Development Environment
+
+![Development Environment configured to be 1.6 by default](/img/blog/2024-04-10-extended-attributes/dbt_version.png)
```
Lightbox?

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+![Development Environment configured to be 1.6 by default](/img/blog/2024-04-10-extended-attributes/dbt_version.png)
+
+The Development Environment is configured to be 1.6 by default.
```
```suggestion
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+The Development Environment is configured to be 1.6 by default.
+
+### Dave’s Development connection settings
```
```suggestion
### Development connection settings
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+### Dave’s Development connection settings
+
+![Dave's development version override](/img/blog/2024-04-10-extended-attributes/dave_version.png)
```
Lightbox?

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+![Dave's development version override](/img/blog/2024-04-10-extended-attributes/dave_version.png)
+
+## Launch Special
```
```suggestion
## Launch special
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+## Launch Special
+
+Each connection you make from every environment is now unique. You can deploy, develop, and test your data with a setup that molds to your organization, not to what’s available in dbt Cloud. Whether you’re looking to create advanced, layered environments to launch new models safely, or enable greater independence between developers, dbt Cloud extends to support what you need. The best part is, we ...
```
```suggestion
Each connection you make from every environment is now unique. You can deploy, develop, and test your data with a setup that molds to your organization, not to what’s available in dbt Cloud. Whether you’re looking to create advanced, layered environments to launch new models safely or enable greater independence between developers, dbt Cloud extends to support what you need. The best part is, we're just getting started: the upcoming _global connections_ feature set will take this even further, allowing you to set multiple connections globally and reuse them wherever needed.
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+
+Each connection you make from every environment is now unique. You can deploy, develop, and test your data with a setup that molds to your organization, not to what’s available in dbt Cloud. Whether you’re looking to create advanced, layered environments to launch new models safely, or enable greater independence between developers, dbt Cloud extends to support what you need. The best part is, we ...
+
+I encourage you to take these new features for a spin by creating a Staging environment, configuring the unique connections you need to enable it at your org, and seeing how it can make your data team more efficient and secure. As always, if you need help or have questions, the [dbt Community Forum](https://discourse.getdbt.com/) and [Slack](https://www.getdbt.com/community/join-the-community) ar ...
```
```suggestion
I encourage you to take these new features for a spin by creating a staging environment, configuring the unique connections you need to enable it at your org, and seeing how it can make your data team more efficient and secure. As always, if you need help or have questions, the [dbt Community Forum](https://discourse.getdbt.com/) and [Slack](https://www.getdbt.com/community/join-the-community) are here to support you. Happy modeling!
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+Lastly let’s consider a more specialized use case. Imagine we have a ‘tiger team’ (consisting of a lone analytics engineer named Dave) tasked with upgrading from dbt version 1.6 to the new ‘Keep on latest version’ setting, to take advantage of added stability and feature access. We want to keep the rest of the data team being productive in dbt 1.6 for the time being, while enabling Dave to upgrad ...
+
+### The Development Environment
+
```
```suggestion
By default, the development environment is configured to be version 1.6:
```

nghi-ly commented on website/blog/2024-04-22-extended-attributes.md:
```diff
@@ -0,0 +1,103 @@
+The Development Environment is configured to be 1.6 by default.
+
+### Dave’s Development connection settings
+
```
```suggestion
Dave's development connection settings are:

```

nghi-ly reviewed (approved):
looks great! mostly minor suggs. all are non-blocking so approving.

gwenwindflower committed:
Apply edits from code review

Awesome editing job by Ly, folding in all the suggestions except for the Lightbox ones, which I'll do a pass of in my local editor.

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

gwenwindflower committed:
Merge branch 'current' into feat/add-eat-blog

gwenwindflower committed:
Update images with Lightbox component
